lib/index.js

```diff
@@ -249,10 +249,17 @@ function promisifyAll(obj) {
  * promise is rejected with that collection. Original indices of the list
  * are preserved.
  *
+ * If the `settle` property in the `options` parameter is set, the returned
+ * list won't have its original indices preserved. Instead, the undefined
+ * indices will be removed.
+ *
  * @param {morePromises~list} list
+ * @param {Object} [options=false]
```

Comment:
Why not leave off the false bit? You never treat it as false. When not set you treat it as an empty object, but I don't know if you really want to document that.

If you do want to document, use this syntax:

    /**
     * @typedef {Object} morePromises~settleOptions
     * @property {boolean} [sparse=false]
     */

and then

    @param {morePromises~settleOptions} [options]